test/e2e/autotls/domain_mapping_test.go

```diff
@@ -0,0 +1,133 @@
+// +build e2e
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autotls
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/networking/test/conformance/ingress"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/reconciler"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+type dmConfig struct {
+	// TLSServiceName is the name of testing Knative Service.
+	// It is not required for self-signed CA or for the HTTP01 challenge when wildcard domain
+	// is mapped to the Ingress IP.
+	TLSServiceName string `envconfig:"tls_service_name" required:"false"`
+	// CustomDomainSuffix is the custom domain used for the domainMapping.
+	CustomDomainSuffix string `envconfig:"custom_domain_suffix" required:"false"`
+}
+
+func TestDomainMappingAutoTLS(t *testing.T) {
+	t.Parallel()
+
+	var env dmConfig
+	if err := envconfig.Process("", &env); err != nil {
+		t.Fatalf("Failed to process environment variable: %v", err)
+	}
+
+	ctx := context.Background()
+
+	clients := e2e.SetupWithNamespace(t, test.TLSNamespace)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "runtime",
+	}
+
+	if len(env.TLSServiceName) != 0 {
+		names.Service = env.TLSServiceName + "-dmtls"
+	}
+
+	// Clean up on test failure or interrupt.
+	test.EnsureTearDown(t, clients, &names)
+
+	// Set up initial Service.
+	svc, err := v1test.CreateServiceReady(t, clients, &names,
+		func(service *v1.Service) {
+			service.Annotations = map[string]string{"networking.knative.dev/disableAutoTLS": "True"}
+		})
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %q: %v", names.Service, err)
+	}
+
+	// Using fixed hostnames can lead to conflicts when multiple tests run at
+	// once, so include the svc name to avoid collisions.
+	suffix := "example.com"
+
+	if env.CustomDomainSuffix != "" {
+		suffix = env.CustomDomainSuffix
+	}
+
+	host := "dm." + suffix
```

Comment:
I think probably I misled you about the host.

So the DNS record with host name "*.${TLS_TEST_NAMESPACE}.${CUSTOM_DOMAIN_SUFFIX}" is added to the DNS server for the DNS configuration: https://github.com/knative/serving/blob/384472192f75797ddb0a51541d1bc0bbb1454954/test/e2e-auto-tls-tests.sh#L89

So the host of DomainMapping needs to be "dm.${TLS_TEST_NAMESPACE}.${CUSTOM_DOMAIN_SUFFIX}"